File: get_initials/initials.rb

def getInitials(name)
    (0..name.length).each do |i|
        if name[i] != 32.chr && (name[i] == name[0] || name[i-1] == 32.chr)
            print(name[i])
        end
    end
    print("\n")
end

print("Enter your full name: ")
name = gets.chomp.strip.to_s

getInitials(name)

File: get_initials/initials.py

def getInitials(name):
    for i in range(len(name)):
        if name[i] != chr(32) and (name[i-1] == chr(32) or name[i] == name[0]):
            print(name[i],end="")

    print("\n")

print("Type your name: ", end="")
name = input()

getInitials(name)
